pkg/gateway/middleware/subdomain_test.go

```diff
@@ -222,20 +222,20 @@ func TestParseSubdomainFields(t *testing.T) {
 		{
 			subdomain: "app1-9a7dbcc",
 			expected: &SubdomainFields{
-				StubGroup: "app1-9a7dbcc",
+				Subdomain: "app1-9a7dbcc",
```

Comment:
This doesn't have a version. Should the version of this also be latest?

Reply:
No because parseSubdomainFields() doesn't set defaults. It only extracts named groups from the subdomain regex.

So in this case, because version doesn't exist in the subdomain when its parsed, the Version field will be an empty string. This is eventually passed to getStubForSubdomain(). In there, the version is parsed and 0 will be the default version if its not parseable. 

From there, the query defined in GetDeploymentBySubdomain has a case statment that will check if the version is 0. If so, it will find the latest version on the deployment CTE/subquery. If its not 0, it'll use the version provided.

Reply:
Sorry, so technically parseSubdomainFields() does set defaults, but they're empty strings.

Thinking about this more, it could make more sense to move parseVersion(fields.Version) out of getStubForSubdomain() and to the parseSubdomainFields() function. Mostly to keep parsing logic closer together. What do you think?

Reply:
I tested this out and it does seem to work well. The one major difference is that Version would be an uint and if its parsed as 0 (meaning latest), then we will always look for and invoke a handler that has "latest" in the path e.g. `/endpoint/:name/latest`.

For example: 

```go
g.GET("/:deploymentName", handler)           // no longer invoked by the middleware
g.GET("/:deploymentName/latest", handler)
g.GET("/:deploymentName/v:version", handler)
```

This isn't an issue, just pointing the differences out. Otherwise, the functionality seems to work the same, but the handler path is more explicit.

Last thing to note is the buildHandlerPath function will have to be updated to handle the Version uint.

```
func buildHandlerPath(stub *types.Stub, fields *SubdomainFields, extraPaths ...string) string {
	pathSegments := []string{"/" + stub.Type.Kind()}

	if stubConfig, err := stub.UnmarshalConfig(); err == nil && !stubConfig.Authorized {
		pathSegments = append(pathSegments, "public", stub.ExternalId)
	} else if fields.StubId != "" {
		pathSegments = append(pathSegments, "id", fields.StubId)
	} else {
		pathSegments = append(pathSegments, fields.Name)

		if fields.Version > 0 {
			pathSegments = append(pathSegments, fmt.Sprintf("v%d", fields.Version))
		} else {
			pathSegments = append(pathSegments, "latest")
		}
	}

	pathSegments = append(pathSegments, extraPaths...)
	return path.Join(pathSegments...)
}
```

I think I like this better too. Mostly for its explicitness in calling a handler with a :latest path param.

Reply:
Just pushed this change.